cosmwasm/contracts/crosschain-swaps/src/execute.rs

```diff
@@ -253,3 +253,236 @@ pub fn recover(deps: DepsMut, sender: Addr) -> Result<Response, ContractError> {
     });
     Ok(Response::new().add_messages(msgs))
 }
+
+/// Set a prefix->channel map in the registry
+pub fn set_channel(
+    deps: DepsMut,
+    sender: Addr,
+    prefix: String,
+    channel: String,
+) -> Result<Response, ContractError> {
+    check_is_contract_owner(deps.as_ref(), sender)?;
+    CHANNEL_MAP.save(deps.storage, &prefix, &channel)?;
+    Ok(Response::new().add_attribute("method", "set_channel"))
+}
+
+/// Remove a prefix from the registry
+pub fn remove_channel(
+    deps: DepsMut,
+    sender: Addr,
+    prefix: String,
+) -> Result<Response, ContractError> {
+    check_is_contract_owner(deps.as_ref(), sender)?;
+    CHANNEL_MAP.remove(deps.storage, &prefix);
```

Comment:
Hrmm, I feel a bit concerned about letting prefix be rebound. I feel like if its bound once, it can be disabled, but shouldn't be editable.

Reply:
I'm thinking of a case like luna, where a new channel is created for the same prefix

Reply:
btw, this function is deleting the prefix=>channel mapping. 

The set_channel msg does reset it: if it exists, it overrides the existing one. We could change that so that we can only set non-existing prefixes. 

I'm happy with that solution, even if I don't see a problem with having it be more flexible

Reply:
I think rebinding should need more governance authority (since you can get rug assumptions or safety pulled out from under you)

Can we do Set New + Disable/Re-enable existing? (We can model Disabled as a field in the value type in the same map)

Reply:
Initially only governance can run these, but another option would be to have two governors so even if we transferred base ownershio to, say, a dao, only gov could rebind. 

I'll change it to set_new+disable/re-enable for now and we can discuss rebinding later.

Reply:
That makes sense!